packages/frontend/lib/index.ts

```diff
@@ -128,7 +130,11 @@ export default class Nango {
             return this.customAuth(providerConfigKey, connectionId, this.convertCredentialsToConfig(credentials), connectionConfig);
         }
 
-        const url = this.hostBaseUrl + `/oauth/connect/${providerConfigKey}${this.toQueryString(connectionId, options as ConnectionConfig)}`;
+        let url = this.hostBaseUrl + `/oauth/connect/${providerConfigKey}${this.toQueryString(connectionId, options as ConnectionConfig)}`;
+
+        if (options && 'credentials' in options && 'token_id' in options.credentials && 'token_secret' in options.credentials) {
+            url = this.hostBaseUrl + `/auth/tba/${providerConfigKey}${this.toQueryString(connectionId, options as ConnectionConfig)}`;
+        }
```

Comment:
I feel the logic and condition above should leave in credentials, for example a`isCustomAuth()` and `urlPath()` functions would significantly simplify the code in `nango.auth()`

Reply:
Yeah, agreed. Let me get this out and refactor this some tomorrow.